vissl/data/vissl_dataset_base.py

```diff
@@ -0,0 +1,44 @@
+from typing import Callable, Dict, Set
+
+import torch
+from torch.utils.data import Dataset
+from vissl.config import AttrDict
+
+
+class VisslDatasetBase(Dataset):
```

Comment:
nit: consider using `abc` to define the abstract methods

```
class Base(abc.ABC):
   @abc.abstractmethod
   def f(self):
      ...
```

This makes the class and sub class not instantiable if they do not define the missing functions

Reply:
Yeah, I wasn't sure whether or not we needed the datasets to inherit from the torch Dataset, I'll test.